tests/cms/views/interactions/test_vote_action.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+from django.contrib.messages.storage.fallback import FallbackStorage
+from django.test import RequestFactory, TestCase
+
+from integreat_compass.cms.models import User, Vote
+from integreat_compass.cms.views.interactions import add_vote
+
+
+def get_request():
+    request = RequestFactory().post("", {"approval": "False"})
+    request.user = User.objects.get(pk=1)
+    setattr(request, "session", "session")
+    messages = FallbackStorage(request)
+    setattr(request, "_messages", messages)
+    return request
+
+
+@pytest.mark.django_db
+def test_add_vote_if_not_exists(load_test_data):
+    assert Vote.objects.filter(creator=1).filter(offer_version=2).count() == 0
+    request = get_request()
+    add_vote(request, 2)
+    assert Vote.objects.filter(creator=1).filter(offer_version=2).count() == 1
+    assert not Vote.objects.filter(creator=1).filter(offer_version=2)[0].approval
```

Comment:
```suggestion
    assert not Vote.objects.get(creator=1, offer_version=2).approval
```
Use `get()` if you expect exactly one matching object and want an error to be thrown if none (or multiple) exist.

Reply:
I thought writing `count() == 1` is more clear ;)

Reply:
Hmm, I disagree. You use `count() == 1` when you want to ensure that the query returns exactly one element without caring about what the element actually is. In this case, this is not true, since we also are interested in the attributes of the returned object (in this case the `approved` field). If you want to check existence and approval in two separate statements, I would prefer the following syntax:
```
vote = Vote.objects.get(creator=1, offer_version=2)
assert not vote.approval
```